data.md

```diff
@@ -3,9 +3,24 @@ WebAssembly Data
 
 ```k
 require "domains.k"
+```
+
+`WASM-SYNTAX` module is used to handle the syntax that is used only for parsing programs but not for parsing rules.
+In WebAssembly string is defined differently with K built-in strings, so we have to write the definition of WebAssembly `WasmString` in a separate module, and use the module just for parsing the program.
+Note that you cannot use a normal K `String` in any production definitions, because the K tokenizer does not support parsing ambiguity.
+
+```k
+module WASM-SYNTAX
+    imports WASM-DATA
+    syntax WasmString ::= r"\\\"(([^\\\"\\\\])|(\\\\[0-9a-fA-F]{2})|(\\\\t)|(\\\\n)|(\\\\r)|(\\\\\\\")|(\\\\')|(\\\\\\\\)|(\\\\u\\{[0-9a-fA-F]{4}\\}))*\\\"" [token]
```

Comment:
See page 108 and 106 re Unicode escape:
It should be `(\\\\u\\{[0-9a-fA-F]+\\})` or `(\\\\u\\{[0-9a-fA-F][0-9a-fA-F]*\\})`